shipyard/src/test/java/org/jclouds/shipyard/features/ServiceKeysApiLiveTest.java

```diff
@@ -0,0 +1,77 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.shipyard.features;
+
+import java.util.List;
+import java.util.UUID;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertNotNull;
+
+import org.jclouds.http.HttpResponseException;
+import org.jclouds.shipyard.domain.servicekeys.CreateServiceKey;
+import org.jclouds.shipyard.domain.servicekeys.DeleteServiceKey;
+import org.jclouds.shipyard.domain.servicekeys.ServiceKey;
+import org.jclouds.shipyard.internal.BaseShipyardApiLiveTest;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+@Test(groups = "live", testName = "ServiceKeysApiLiveTest", singleThreaded = true)
+public class ServiceKeysApiLiveTest extends BaseShipyardApiLiveTest {
+
+   private final String serviceKeyDescription = "ShipyardJCloudsLiveTest";
+   private String serviceKey = null;
+   
+   @AfterClass (alwaysRun = true)
+   protected void tearDown() {
+      assertNotNull(serviceKey, "Expected serviceKey to be set but was not");
+      DeleteServiceKey deleteServiceKey = DeleteServiceKey.create(serviceKey);
+      api().deleteServiceKey(deleteServiceKey);
+   }
+   
+   public void testCreateServiceKey() throws Exception {
+     CreateServiceKey createServiceKey = CreateServiceKey.create(serviceKeyDescription);
+     ServiceKey possibleServiceKey = api().createServiceKey(createServiceKey);
+     assertNotNull(possibleServiceKey, "Did not successfully create ServiceKey");
+     assertTrue(possibleServiceKey.description().equals(serviceKeyDescription), "ServiceKey description returned from Shipyard did not match expected value");
+     serviceKey = possibleServiceKey.key();
+   }
+   
+   @Test (dependsOnMethods = "testCreateServiceKey")
+   public void testListServiceKeys() throws Exception {
+      List<ServiceKey> possibleServiceKeys = api().listServiceKeys();
+      assertNotNull(possibleServiceKeys, "possibleServiceKeys was not set");
+      assertTrue(possibleServiceKeys.size() > 0, "Expected at least 1 ServiceKey but list was empty");
+      boolean serviceKeyFound = false;
+      for (ServiceKey possibleKey : possibleServiceKeys) {
+         if (possibleKey.key().equals(serviceKey)) {
+            serviceKeyFound = true;
+         }
+      }
+      assertTrue(serviceKeyFound, "Expected but could not find ServiceKey amongst " + possibleServiceKeys.size() + " found");
+   }
+   
+   @Test (expectedExceptions = HttpResponseException.class)
+   public void testRemoveNonExistentServiceKey() throws Exception {
+      DeleteServiceKey deleteServiceKey = DeleteServiceKey.create(UUID.randomUUID().toString().replaceAll("-", ""));
+      api().deleteServiceKey(deleteServiceKey);
```

Comment:
Given the exception from the provider can you determine if it is a "Not Found" exception? If you can, I'd create a custom fallback or configure the error handler to propagate a `ResourceNotFoundException` instead of the generic `HttpResponseException`. That's how all other jclouds providers behave.